### PR TITLE
Thai/check owner eq user

### PR DESCRIPTION
**Pull request by athai**
Checks that the user logged in is the same as the user listed in the JSON object. A test checking whether the user making the post has admin. rights will need to be added later.

### CODE REVIEW

**Comment by tschuy** [comment]
Looks good to me! 

@ElijahCaine can we get another +1 or some feedback?

File: src/helpers.js [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
-module.exports = {
+var app = require('./app');
+var knex = app.get('knex');
 
+module.exports = {
+    checkUser: function(username, objUser) {
+        return new Promise(function(resolve, reject) {
+            if (username === objUser) {
+                knex('users').select('id')
+                .where('username', username).then(function(users) {
```
Comment (tschuy) [inline src/helpers.js]: where 'username' is objUser, though this won't come up until we decide on using active users or not.

File: src/helpers.js [hunk 2]
```diff
@@ -1,5 +1,23 @@
 // src/helpers.js
 
-module.exports = {
+var app = require('./app');
```
Comment (iankronquist) [inline src/helpers.js]: Won't this start up a server when you import it?
I think you'll want to use the export a function pattern: https://github.com/osuosl/timesync-node/blob/develop/src/times.js#L1

File: src/helpers.js [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+var knex = app.get('knex');
 
+module.exports = {
+    checkUser: function(username, authUser) {
+        return new Promise(function(resolve, reject) {
+            if (username === authUser) {
+                knex('users').select('id')
+                .where('username', username).then(function(users) {
+                    /* Knex only returns lists and not single items. 'users'
```
Comment (iankronquist) [inline src/helpers.js]: You want this: http://knexjs.org/#Builder-first

Comment (tschuy) [inline src/helpers.js]: TIL that's a thing. Huh.

File: src/helpers.js (hunk 2 above)
Comment (tschuy) [inline src/helpers.js]: I mean, I've not seen any extra "app now listening on"s pop up when we do this, and this isn' t the first place we've done it.

Comment (iankronquist) [inline src/helpers.js]: I know that some testing frameworks swallow stdout so that's not exactly proof of anything.

Comment (tschuy) [inline src/helpers.js]: mocha does not. I use console.log development with mocha alllll the time.

Comment (iankronquist) [inline src/helpers.js]: Ah, you're now exporting the app in `app.js`. I think that changes things. In any case this differs from what we do elsewhere and I think we should be consistent.

Comment (athai) [inline src/helpers.js]: How should this be changed then?

Comment (tschuy) [inline src/helpers.js]: I don't believe it has to be chagned.

Comment (iankronquist) [inline src/helpers.js]: I think either this should be changed to use the `function(app)` pattern as linked above or we should abandon that for this approach. It's not strictly necessary but it looks fairly easy to pull off.

**Comment by iankronquist** [comment]
`@athai 	The linter cares too much` ~ That is an excellent commit message.

**Comment by tschuy** [comment]
This looks good to me! nice job :)

File: tests/helpers.js [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
-    // Remove it when you write real tests.
-    expect(helper);
+    describe('checkUser', function() {
+        it('Returns user ID if username == user', function(done) {
+            helpers.checkUser('tschuy', 'tschuy').then(function(userID) {
+                expect(userID).to.equal(2);
+                done();
+            });
+        });
+        // Later: Include a test that checks if username is admin
```
Comment (ElijahCaine) [inline tests/helpers.js]: Should we add this test now or is it a feature that will be implemented later (along with corresponding tests)?

Comment (athai) [inline tests/helpers.js]: Admin rights is a feature that will be implemented later. The comment is just acting as a placeholder/reminder.

**Comment by ElijahCaine** [comment]
Beyond that one nit-pick about adding the admin test, this gets a +1 from me.

**Comment by ElijahCaine** [comment]
Great. Looks good to me.

**official :+1:**

**Comment by tschuy** [comment]
This looks good to me :)

**Pull request merged**